### PR TITLE
Replace apigen with Sami in CI

### PR DESCRIPTION
**Pull request by williamdes**
- Removed apigen/apigen
- Added sami/sami
- Generated a configuration and added it in /test/ as config-sami.php
- Removed "apidocs are allowed to fail" in .travis.yml

Closes: #14345

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -82,8 +82,6 @@ matrix:
     - php: "nightly"
       env: CI_MODE=apidocs
   allow_failures:
-    - php: "7.1"
-      env: CI_MODE=apidocs
```
Comment (mauriciofauth) [inline .travis.yml]: You can remove it from the matrix as well. See 6101eea28540cb3d613e589e4e35cec49d3b7537.

File: composer.json [hunk 2]
```diff
@@ -87,7 +87,8 @@
         "pragmarx/google2fa": "^3.0",
         "bacon/bacon-qr-code": "^1.0",
         "samyoul/u2f-php-server": "^1.1",
-        "phpmyadmin/coding-standard": "^0.3"
+        "phpmyadmin/coding-standard": "^0.3",
+        "sami/sami": "^4.0"
```
Comment (mauriciofauth) [inline composer.json]: Sami's readme doesn't recommend installing Sami as a dependency in Composer. Would not it be better to install it separately?

**Review by MauricioFauth — COMMENTED** [review]
Thank you for your contribution.

**Comment by williamdes** [comment]
OSX build failed for no good reason :laughing:

**Comment by mauriciofauth** [comment]
Merged, thanks for your contribution!

**Pull request merged**